napari/layers/labels/labels.py

```diff
@@ -665,6 +679,44 @@ def fill(self, coord, new_label, refresh=True):
         if refresh is True:
             self.refresh()
 
+    @lru_cache(maxsize=64)
+    def sphere_indices(self, radius, sphere_dims):
+        """Generate centered indices within circle or n-dim sphere.
+
+
+        Parameters
+        -------
+        radius : float
+            Radius of circle/sphere
+        sphere_dims : int
+            Number of circle/sphere dimensions
+
+        Returns
+        -------
+        mask_indices : array
+            Centered indices within circle/sphere
+        """
+        # Create multi-dimensional grid to check for
+        # circle/membership around center
+        vol_radius = radius + 0.5
+
+        sliced_index = [
+            slice(-vol_radius, vol_radius + 1) for __ in range(sphere_dims)
+        ]
+        sliced_dist = [
+            slice(-vol_radius, vol_radius + 1) for __ in range(sphere_dims)
+        ]
+
+        indices = np.mgrid[sliced_index].T.reshape(-1, sphere_dims)
+        distances = np.mgrid[sliced_dist].T.reshape(-1, sphere_dims)
+
+        distances = distances ** 2
+        distances = np.sqrt(np.sum(distances, axis=1))
+        # Use distances within desired radius to mask indices in grid
+        mask_indices = indices[distances <= radius].astype(int)
```

Comment:
This can be simplified quite a bit, and sped up by avoiding a square root:

```suggestion
        indices_slice = [slice(-vol_radius, vol_radius + 1),] * sphere_dims
        indices = np.mgrid[indices_slice].T.reshape(-1, sphere_dims)
        distances_sq = np.sum(indices ** 2, axis=1)
        # Use distances within desired radius to mask indices in grid
        mask_indices = indices[distances_sq <= radius ** 2].astype(int)
```

Reply:
Ah true! I never fixed that up after refactoring to the function